surfactant/infoextractors/file_decompression.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2025 Lawrence Livermore National Security, LLC
+# See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: MIT
+# Copyright 2025 Lawrence Livermore National Security, LLC
+# See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: MIT
+import tarfile
+import tempfile
+import zipfile
+from queue import Queue
+from typing import Any, Dict, Optional
+
+from loguru import logger
+
+import surfactant.plugin
+from surfactant import ContextEntry
+from surfactant.sbomtypes import SBOM, Software
+
+
+def is_compressed(filename):
+    # Map file extensions to their compression formats
+    compression_formats = {
+        ".zip": "zip",
+        ".tar": "tar",
+        ".tar.gz": "tar.gz",
+        ".tar.bz2": "tar.bz2",
+        ".tar.xz": "tar.xz",
+    }
+    for ext, format in compression_formats.items():
+        if filename.endswith(ext):
+            return format
```

Comment:
```suggestion
    for ext, fmt in compression_formats.items():
        if filename.endswith(ext):
            return fmt
```

Fix pylint: `surfactant/infoextractors/file_decompression.py:31:13: W0622: Redefining built-in 'format' (redefined-builtin)`